deployments/deploy.ps1

```diff
@@ -1,50 +1,47 @@
-try {
-    Import-Module .\powerform\pf-deployment-context\pf-deployment-context.psd1
-    Import-Module .\azure\resource-containers\pf-management-group\pf-management-group.psd1
-    Import-Module .\azure\resource-containers\pf-subscription\pf-subscription.psd1
-    Import-Module .\azure\resource-containers\pf-resource-group\pf-resource-group.psd1
-    Import-Module .\azure\resources\pf-key-vault\pf-key-vault.psd1
-    Import-Module .\azure\resources\pf-log-analytics\pf-log-analytics.psd1
+#try {
+    $VerbosePreference = "continue"
+    Import-Module .\azure\pf-azure.psd1
 
-    #$pfContext = New-PfDeploymentContext
     $currentDir = Split-Path -Parent $MyInvocation.MyCommand.Path
-    #$pfContext = New-PfDeploymentContext -AzConfigFile "$($currentDir)\config\az-config.json"
-    $pfContext = New-PfDeploymentContext -GlobalConfigFile "$($currentDir)\config\global-config.json" -AzConfigFile "$($currentDir)\config\az-config.json"
-
-    Set-PfAzureContext -CompanyInfo $pfContext.GlobalConfig.CompanyInfo `
-        -Group $pfContext.GlobalConfig.Groups.Lurie `
-        -Label $pfContext.GlobalConfig.Labels.WebTeam `
-        -Environment $pfContext.GlobalConfig.Environments.Qa `
-        -AzRegion $pfContext.AzConfig.Regions.CentralUs
-
-    #Set-PfAzureContext -SubscriptionName "MYAEA-KKZH-D" -ResourceGroupName "RG-MYAEA-KKZH-ADM-D" -AzRegion $pfContext.AzConfig.Regions.CentralUs
-    <#Set-PfAzureContext -CompanyInfo $pfContext.GlobalConfig.CompanyInfo `
-        -Group $pfContext.GlobalConfig.Groups.Team1 `
-        -Label $pfContext.GlobalConfig.Labels.Networking `
-        -Environment $pfContext.GlobalConfig.Environments.Development `
-        -AzRegion $pfContext.AzConfig.Regions.CentralUs
-        #>
-    #Set-PfAzureContext  -SubscriptionName "MYAEA-KKZH-D" -ResourceGroupName "RG-MYAEA-KKZH-ADM-D" -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -Label "ADM" -EnvironmentLetter "D" -AzRegion $pfContext.AzConfig.Regions.CentralUs
+    $pfConfig = Initialize-PowerForm -CustomConfigFile "$($currentDir)\config\custom-config.json"
 
+    $pfContext = New-PfDeploymentContext -Name "OcwDevelopment"
+    
+    Add-PfBuildContext -Name "LurieAdminComponentsDev" `
+        -Company $pfConfig.Companies.OpenCaseWork `
+        -Group $pfConfig.Groups.Lurie `
+        -Label $pfConfig.Labels.AdminTeam `
+        -Environment $pfConfig.Environments.Development `
+        -Region $pfConfig.Regions.CentralUs
+    
+    Set-PfBuildContext -Name "LurieAdminComponentsDev"
     #Login-AzAccount
-    $rootMg = New-PfManagementGroup -DeploymentContext $pfContext
-    #$pfContext.AddDefinition($rootMg)
-
-    $getRootMg = Get-PfManagementGroup -Name $rootMg.Options.Name
+    $rootMg = New-PfManagementGroup
+    $rootMg.Save()
```

Comment:
So now after you create a new definition you have to call the "save" method to save it to the context and do any dependency tracking on it.  Thoughts?

Reply:
As I'm progressively thinking more and more functional thinking leaning towards flipping around to a Save or Set that takes in the MG Definition

Reply:
now i recall we were also talking about just implicitly handling save...